microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -47,6 +55,164 @@ class VmSnapsotLinuxBVTExtension(TestSuite):
+
+    @TestCaseMetadata(
+        description="""
+        Runs the custom script extension and verifies it execution on the
+        remote machine.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension],
+        ),
+    )
+    def verify_exclude_disk(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        result: TestResult,
+    ) -> None:
+        # Any extension will do, use CustomScript for convenience.
+        # Copy the local files into the VM.
+        # Move the files to the extension directory
+        # Give the execution permissions to the files
+        # Run the files
+        # Validate the results
+        extension_dir = self.find_extension_dir(node)
+        if extension_dir == "":
+            self._verify_vmsnapshot_extension(log, node, environment)
+            trial = 0
+            while trial < 3:
+                time.sleep(2)
+                extension_dir = self.find_extension_dir(node)
+                if extension_dir == "":
+                    trial = trial + 1
+                else:
+                    break
+
+        # installing all the required packages
+        posix_os: Posix = cast(Posix, node.os)
+        posix_os.install_packages("python2.7")
+        posix_os.install_packages("pip")
+        url = "https://bootstrap.pypa.io/pip/2.7/get-pip.py"
+        args = "-o get-pip.py"
+        curl = Curl(node)
+        curl.fetch(url=url, arg=args, shell=True, execute_arg="", sudo=True)
+        node.execute(cmd="python2.7 get-pip.py", shell=True, sudo=True)
+        node.execute(cmd="python2.7 -m pip --version")
+        node.execute(cmd="python2.7 -m pip install mock", sudo=True)
+        # copy the file into the vm
+        self.copy_to_node(node, "handle.txt")
+
+        if extension_dir != "":
+            # moving the handle.py file to extension directory
+            script = (
+                "#!/bin/sh\n"
+                f"mv {node.working_path}/handle.txt {extension_dir}/handle.py"
+            )
+            script_base64 = base64.b64encode(bytes(script, "utf-8")).decode("utf-8")
+            settings = {"script": script_base64}
+            extension = node.features[AzureExtension]
+            extension.create_or_update(
+                name="CustomScript",
+                publisher="Microsoft.Azure.Extensions",
+                type_="CustomScript",
+                type_handler_version="2.1",
+                auto_upgrade_minor_version=True,
+                settings=settings,
+            )
+            log.info(extension_dir)
+            # give the execution permissions to the file
+            node.execute(
+                cmd=f"chmod 777 {extension_dir}/handle.py", shell=True, sudo=True
+            )
+            # execute the file
+            script_result = node.execute(
+                cmd=f"python2.7 {extension_dir}/handle.py", shell=True, sudo=True
```

Comment:
Any reason for hardcoding the python version ?

Reply:
mock module that we are installing works in python 2.7